### PR TITLE
Require Java 11 and Jenkins 2.361.4 or newer

### PR DESCRIPTION
**Pull request by VignanMudedla**
1. updated parent pom to 4.59(allow compilation with java 11)
2. updated base Jenkins version to 2.361.4

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

**Comment by VignanMudedla** [comment]
Hey @MarkEWaite! Can you help we with the windows-8 build fail.
I'm new to Open-source and this is my first PR. I Started contributing to Jenkins by watching " modernising the Jenkins Plugin tutorial"

I'll be more than happy if any one could help me with this

**Comment by MarkEWaite** [comment]
It needs two more steps:

1. Update the Jenkinsfile to replace Java 8 with Java 11 and Java 17 as described in ["Add a Jenkinsfile"](https://www.jenkins.io/doc/developer/tutorial-improve/add-a-jenkinsfile/#create-the-jenkinsfile)
2. Replay the ci.jenkins.io job with the updated Jenkinsfile (needs  a ci.jenkins.io admin like me to do that)

**Comment by MarkEWaite** [comment]
> Hey @MarkEWaite is this good?
> 
> Screenshot 2023-04-13 at 8 49 32 PM
> 
> can I add another new configuration for windows-17?

Because the test configurations incur a cost, we increase test configurations only when we believe that the additional configuration will give useful information for the many times that it will be run (each pull request runs the Pipeline, master branch runs the Pipeline, etc.)

The "Improve a plugin" tutorial chooses only two configurations because that felt like the best balance between information and cost.  One of the configurations is Linux and one is Windows because Windows and Linux have some very important differences.  One of the configurations is Java 11 and one is Java 17 because there are some interesting differences between those two.  So long as we touch each of those key differences in at least one configuration, that gives us most of the benefit without the cost of testing all combinations of the configurations.  We're using [pairwise testing](https://en.wikipedia.org/wiki/All-pairs_testing) in the Jenkinsfile to cover the most important cases without needing to perform exhaustive testing of all combinations

**Comment by VignanMudedla** [comment]
Hey @MarkEWaite I've updated Jenkinsfile to java 11 and java 17 but Jenkins CI is yet building for java 8. build failed again.
I also updated Jenkins version for linux-17 to 2.361.4
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fjavadoc-plugin/detail/PR-96/3/pipeline/

sorry, I'm pinging you for little things but I really need some help

**Comment by MarkEWaite** [comment]
> sorry, I'm pinging you for little things but I really need some help

A change to the Jenkinsfile is one of those things that requires a ping of someone else unless you are a maintainer of the plugin.  I'll start a build of the plugin with your modified Jenkinsfile so that you can see the results.

File: Jenkinsfile
Comment (MarkEWaite) [inline Jenkinsfile:11]: Usually you want to test a different Jenkins version than the default so that you know if other problems are detected by the newer Jenkins version
```suggestion
                [ platform: "linux", jdk: "17", jenkins: '2.387.2' ]
```

**Review by MarkEWaite — COMMENTED** [review]
Optional comment on the Jenkins version being tested

File: Jenkinsfile
Comment (VignanMudedla) [reply Jenkinsfile]: may I know why you've changed the Jenkins version to 2.387.2?
I just updated the Jenkins version to the recommended one in the documentation.

@MarkEWaite But yet the check is failed

Comment (MarkEWaite) [reply Jenkinsfile]: > may I know why you've changed the Jenkins version to 2.387.2?

The Jenkins version 2.361.4 is the value that is used already in the line that tests Jenkins on Linux.  The `jenkins.version` value is the default when no `jenkins:` parameter is defined.

I used a newer version (2.400) in the most recent build so that it would test a wider range of functionality with the same build process.

**Comment by VignanMudedla** [comment]
Hey @MarkEWaite Thanks a lot for your support!!
Can you review this PR and [Deeper spotbugs checks](https://github.com/jenkinsci/javadoc-plugin/pull/97) PR ?

File: Jenkinsfile
Comment (MarkEWaite) [inline Jenkinsfile:6]: Don't need to also run on Window Java 11.  One platform for Java 11 is enough.
```suggestion
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Limit the number of platforms to reduce CI cost and assure fast evaluation of pull requests

File: Jenkinsfile
Comment (MarkEWaite) [inline Jenkinsfile:9]: Sorry that I didn't comment on this earlier, but these comments are no longer valid.  We merged the Guava and Guice changes a long time ago.
```suggestion
```

File: Jenkinsfile
Comment (VignanMudedla) [reply Jenkinsfile]: > Don't need to also run on Window Java 11. One platform for Java 11 is enough.

Hey @MarkEWaite just committed the recommended changes. waiting for you to reply the CIi job

File: pom.xml
Comment (MarkEWaite) [inline pom.xml:28]: I missed that there is another change needed in this file in order to complete the update of the parent pom and the base Jenkins version.  That [step](https://www.jenkins.io/doc/developer/tutorial-improve/update-base-jenkins-version/#update-minimum-required-jenkins-version) is described in the [tutorial](https://www.jenkins.io/doc/developer/tutorial-improve/update-base-jenkins-version/).

The [plugin bill of materials](https://www.jenkins.io/doc/developer/tutorial-improve/use-plugin-bill-of-materials/) needs to be updated  from `bom-2.303.x` version `1500.ve4d05cd32975` to `bom-2.361.1` version `1981.v17df70e84a_a_1` as shown at https://github.com/jenkinsci/javadoc-plugin/blob/5f78d449fd15977f14b04cada74b090b67e46acd/pom.xml#L35-L36

File: Jenkinsfile
Comment (MarkEWaite) [inline Jenkinsfile:6]: This should be testing Windows rather than Linux so that the tests cover both Windows and Linux and cover both Java 11 and Java 17.  

As a matter of style, we prefer single quotes instead of double quotes in Pipeline strings when the Pipeline string does not include a variable to be interpreted or a method call to be made when the string is computed.  Single quotes in groovy mean that the string is not to be interpolated by the groovy parser.  In this case, the result is exactly the same final string, but single quotes are consistent with the first line and with the example in the [tutorial](https://www.jenkins.io/doc/developer/tutorial-improve/add-a-jenkinsfile/#create-the-jenkinsfile).
```suggestion
                [ platform: 'windows', jdk: '17', jenkins: '2.387.2' ]
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
I have confirmed that the changes to the Jenkinsfile run as expected in a [recent job](https://ci.jenkins.io/job/Plugins/job/javadoc-plugin/view/change-requests/job/PR-96/10/).  I had missed two items in previous reviews.  Included in this review.

File: pom.xml
Comment (VignanMudedla) [reply pom.xml:28]: @MarkEWaite,  You specified In the tutorial that the bom version should be the latest release version but `226.v71211feb_e7e9`  is the latest release --> "correct me if I'm wrong"

Out of my interest, what is the significance of specifying the version in bom?

Comment (MarkEWaite) [reply pom.xml:28]: > the bom version should be the latest release version but 226.v71211feb_e7e9 is the latest release

https://github.com/jenkinsci/bom/releases shows that `1981.v17df70e84a_a_1` is the most recent release of the plugin bill of materials. 

> Out of my interest, what is the significance of specifying the version in bom?

The [tutorial](https://www.jenkins.io/doc/developer/plugin-development/dependency-management/#update-the-jenkins-core-bill-of-materials-bom) includes a [video](https://youtu.be/Fev8KfFsPZE?t=807) that gives an overview of the plugin bill of materials.  

New versions of the plugin bill of materials update the versions of plugins that are defined in the bill of materials.  Newer bill of materials versions allow plugins to depend on newer plugin releases without requiring that the plugin track the version of each plugin in its own pom file.

Bill of materials `bom-2.361.x` version `1500.ve4d05cd32975` will list older plugin versions than `bom-2.361.x` version `1981.v17df70e84a_a_1`.  By using the most recent release of the plugin bill of materials, you're choosing to rely on the most recent releases of plugins as the dependencies of this plugin.

Comment (VignanMudedla) [reply pom.xml:28]: Sorry for that,  I just got confused!!

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
Still needs the bom-2.361.x change and the latest bom version in that change.  When you resolve a comment, I assume it means you addressed the concern, but in this case, the concern was not addressed.  Let's get more benefits from the Java 11 transition, not just update the minimum Jenkins version.

File: pom.xml
Comment (MarkEWaite) [reply pom.xml:28]: No problem at all.  Thanks for being involved!

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Comment by VignanMudedla** [comment]
> Still needs the bom-2.361.x change and the latest bom version in that change. When you resolve a comment, I assume it means you addressed the concern, but in this case, the concern was not addressed. Let's get more benefits from the Java 11 transition, not just update the minimum Jenkins version.

I just committed that change!!
Extremely sorry for marking it resolved before making the change
this is because I'm new to GitHub

**Comment by MarkEWaite** [comment]
> because I'm new to GitHub

Delighted that you're willing to learn and very happy to have your contribution.  Thanks again.  Once this change is merged I have two more changes that I will push through, then the plugin will release so that you can see the results of your efforts in a plugin release.

**Pull request merged**
